src/hitchhiker/tree/messaging.cljc

```diff
@@ -0,0 +1,252 @@
+(ns hitchhiker.tree.messaging
```

Comment:
I have readded the files in their current version, I can still rebase in the end and remove the intermediary commits.